payments.py

```diff
@@ -0,0 +1,40 @@
+from django.contrib.auth.models import Permission, User
+
+from memberships.models import Member, Membership, FailedPayment, Payment
+from funky_time import epoch_to_datetime
+
+
+def handle_stripe_payment(event):
+    if event["type"] == "invoice.payment_failed":
+        FailedPayment.objects.create(
+            stripe_customer_id=event["data"]["object"]["customer"],
+            stripe_subscription_id=event["data"]["object"]["subscription"],
+            stripe_event_type=event["type"],
+        )
```

Comment:
Remeber to always return a `HttpResponse` from this method. A `2xx` if this webhook request was valid / could be handled successfully, and a non `2xx` if it could not.

```suggestion
        FailedPayment.objects.create(
            stripe_customer_id=event["data"]["object"]["customer"],
            stripe_subscription_id=event["data"]["object"]["subscription"],
            stripe_event_type=event["type"],
        )
        return HttpResponse(200)
```

> You will have to import `HttpResponse` for the above to work.

Reply:
So for a successful payment a `2xx` response is great.

For a failed payment, will we want a non `2xx` response so that stripe tries again later?

Reply:
If we receive a `failed_payment` event we will _still_ want to return a 2xx if we have handled it correctly (logged a record of it to the database).

My understanding is that a failed payment is one that cannot be retried and we would need to contact the member ourselves to get them to retry (eg: with a different card).

I believe the docs for failed payments on Stripe are https://stripe.com/docs/declines, but I haven't had a chance to read through them in detail.

Reply:
Yes this makes sense